src/types/config.ts

```diff
@@ -0,0 +1,35 @@
+import { DataSourceJsonData } from '@grafana/data';
+
+export interface CHConfig extends DataSourceJsonData {
+  username: string;
+  server: string;
+  protocol: Protocol;
+  port: number;
+  defaultDatabase?: string;
+  tlsSkipVerify?: boolean;
+  tlsAuth?: boolean;
+  tlsAuthWithCACert?: boolean;
+  secure?: boolean;
+  validate?: boolean;
+  timeout?: string;
+  queryTimeout?: string;
+  customSettings?: CHCustomSetting[];
```

Comment:
It looks like this type is used in the config editor, it will need further changes if using `ReadonlyArray`. I will make note of this when updating the config structure